PacketDotNet/OSPFv2Packet.cs

```diff
@@ -0,0 +1,1056 @@
+                        case LSAType.Router:
+                            ret.Add(new RouterLSA(this.header.Bytes, offset, l.Length));
+                            break;
+                        case LSAType.Summary:
+                        case LSAType.SummaryASBR:
+                            ret.Add(new SummaryLSA(this.header.Bytes, offset, l.Length));
+                            break;
+                    }
+                    offset += l.Length;
+                }
+                return ret;
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder packet = new StringBuilder();
+            packet.Append(base.ToString());
+            packet.Append(" ");
+            packet.AppendFormat("LSANumber: {0} ", this.LSANumber);
+            return packet.ToString();
+        }
+
+        public override string ToString(StringOutputType outputFormat)
+        {
+            return ToString();
+        }
+    }
+
+    /// <summary>
+    /// Link State Acknowledgment Packets are OSPF packet type 5.  To make
+    /// the flooding of LSAs reliable, flooded LSAs are explicitly
+    /// acknowledged. See http://www.ietf.org/rfc/rfc2328.txt for details.
+    /// </summary>
+    public class OSPFv2LSAPacket : OSPFv2Packet
+    {
+        /// <value>
+        /// The packet type
+        /// </value>
+        public static OSPFPacketType packetType = OSPFPacketType.LinkStateAcknowledgment;
+
+        /// <summary>
+        /// Constructs an OSPFv2 Link State Acknowledge packet from ByteArraySegment
+        /// </summary>
+        /// <param name="bas">
+        /// A <see cref="ByteArraySegment"/>
+        /// </param>
+        public OSPFv2LSAPacket(ByteArraySegment bas)
+        {
+            header = new ByteArraySegment(bas.Bytes);
+        }
+
+        /// <summary>
+        /// Constructs an Link OSPFv2 State Acknowledge packet
+        /// </summary>
+        public OSPFv2LSAPacket()
+        {
+            byte[] b = new byte[OSPFv2Fields.LSAHeaderPosition];
+            Array.Copy(header.Bytes, b, header.Bytes.Length);
+            this.header = new ByteArraySegment(b, 0, OSPFv2Fields.LSAHeaderPosition);
+            this.Type = packetType;
+
+            this.PacketLength = (ushort)header.Bytes.Length;
+        }
+
+        /// <summary>
+        /// Constructs an OSPFv2 Link State Acknowledge packet with LSA headers
+        /// </summary>
+        /// <param name="lsas">List of the LSA headers</param>
+        public OSPFv2LSAPacket(List<LSA> lsas)
+        {
+            int length = lsas.Count * OSPFv2Fields.LSAHeaderLength;
+            int offset = OSPFv2Fields.HeaderLength;
+            byte[] bytes = new byte[length + OSPFv2Fields.HeaderLength];
+
+            Array.Copy(header.Bytes, bytes, header.Length);
+            for (int i = 0; i < lsas.Count; i++)
+            {
+                Array.Copy(lsas[i].Bytes, 0, bytes, offset, 20); //20 bytes per header

```

Comment:
can this 20 come from a constant defined in the LSA structure? You could get at it like LSA.Something rather than using 20.